lib/streamlit/runtime/scriptrunner/script_requests.py

```diff
@@ -104,42 +104,47 @@ def request_rerun(self, new_data: RerunData) -> bool:
                 return False
 
             if self._state == ScriptRequestType.CONTINUE:
-                # If we're running, we can handle a rerun request
-                # unconditionally.
+                # The script is currently running, and we haven't received a request to
+                # rerun it as of yet. We can handle a rerun request unconditionally so
+                # just change self._state and set self._rerun_data.
                 self._state = ScriptRequestType.RERUN
                 self._rerun_data = new_data
                 return True
 
             if self._state == ScriptRequestType.RERUN:
-                # If we have an existing Rerun request, we coalesce this
```

Comment:
The code here used to have some optimizations to not make copies of `RerunData` objects when possible, but any performance gains we see from this are likely marginal, so I changed the code to always make a copy of `RerunData` in this `if` clause since it makes things more readable